Assets/Python/RFCUtils.py

```diff
@@ -243,61 +240,43 @@ def killUnitsInArea(self, iPlayer, lPlots):
+		oldCapital = gc.getPlayer(iOldOwner).getCapitalCity()
+		
 		for (x, y) in lPlots:
-			killPlot = gc.getMap().plot(x,y)
+			killPlot = gc.getMap().plot(x, y)
+			if bSkipPlotCity and killPlot.isCity():
+				#print (killPlot.isCity())
+				#print 'do nothing'
+				continue
+			lPlotUnits = []
 			iNumUnitsInAPlot = killPlot.getNumUnits()
 			if iNumUnitsInAPlot > 0:
-				bRevealedZero = False
-				if gc.getMap().plot(0, 67).isRevealed(iNewOwner, False):
-					bRevealedZero = True
 				#print ("killplot", x, y)
-				if bSkipPlotCity and killPlot.isCity():
-					#print (killPlot.isCity())
-					#print 'do nothing'
-					pass
-				else:
-					j = 0
-					oldCapital = gc.getPlayer(iOldOwner).getCapitalCity()
-					for i in range(iNumUnitsInAPlot):
-						unit = killPlot.getUnit(j)
-						#print ("killplot", x, y, unit.getUnitType(), unit.getOwner(), "j", j)
-						if unit.getOwner() == iOldOwner:
-							# Leoreth: Italy shouldn't flip so it doesn't get too strong by absorbing French or German armies attacking Rome
-							if iNewOwner == iItaly and iOldOwner < iNumPlayers:
-								unit.setXYOld(oldCapital.getX(), oldCapital.getY())
-							else:
-								unit.kill(False, iBarbarian)
-								
-								# Leoreth: can't flip naval units anymore
-								if unit.getDomainType() == DomainTypes.DOMAIN_SEA:
-									continue
-									
-								# Leoreth: ignore workers as well
-								if unit.getUnitType() in [iWorker, iPunjabiWorker, iMadeireiro]:
-									continue
-								
-								if not (unit.isFound() and not bKillSettlers) and not unit.isAnimal():
-									self.makeUnit(unit.getUnitType(), iNewOwner, (0, 67), 1)
+				for iUnit in reversed(range(iNumUnitsInAPlot)):
+					unit = killPlot.getUnit(iUnit)
+					#print ("killplot", x, y, unit.getUnitType(), unit.getOwner(), "j", j)
+					if unit.getOwner() == iOldOwner:
+						# Leoreth: Italy shouldn't flip so it doesn't get too strong by absorbing French or German armies attacking Rome
+						if iNewOwner == iItaly and iOldOwner < iNumPlayers:
+							unit.setXYOld(oldCapital.getX(), oldCapital.getY())
 						else:
-							j += 1
-					tempPlot = gc.getMap().plot(0,67)
-					#moves new units back in their place
-					if tempPlot.isUnit():
-						iNumUnitsInAPlot = tempPlot.getNumUnits()
-						for i in range(iNumUnitsInAPlot):
-							unit = tempPlot.getUnit(0)
-							unit.setXYOld(x,y)
-						iCiv = iNewOwner
-						if not bRevealedZero:
-							for (x, y) in self.surroundingPlots((0, 67), 2):
-								gc.getMap().plot(x, y).setRevealed(iCiv, False, True, -1)
-
-
-
+							unit.kill(False, iBarbarian)
+							
+							# Leoreth: can't flip naval units anymore
+							if unit.getDomainType() == DomainTypes.DOMAIN_SEA:
+								continue
+								
+							# Leoreth: ignore workers as well
+							if unit.getUnitType() in [iWorker, iPunjabiWorker, iMadeireiro]:

```

Comment:
Maybe an opportunity to use utils.getBaseUnit(), also should include the Labourer unit.